docs/troubleshooting-guide/my-dashboard-is-slow.md

```diff
@@ -0,0 +1,116 @@
+3. [This guide][admin-caching] explains how to change the minimum query duration (we cache anything that takes longer than that to run) and the maximum cache size for each query result. You may need to experiment with these values over several days to find the best balance. If the problem appears to be caused by a high proportion of sandboxed queries, check that the cache is large enough to store all of their results.
+
+## Are values repeatedly being converted on the fly?
+
+**Root cause:** Low performance can be caused by incorrect typing of columns, e.g., by storing a numeric value as a string.  When this happens, the query converts values on the fly each time it runs. (Even a handwritten query will be slow if it has to do this conversion every time.)
+
+**Steps to take:**
+
+1. Go to Admin > Data Model and look at the raw data types of the columns being used in your question. If (for example) a column is stored as a string when the question needs a number or a timestamp, you may need to convert the column in place.
+2. Alternatively, you can create a new table with the converted values in the right type. This takes more storage, and needs to be updated as new data arrives, but will not affect any other applications that are using the same database.
+
+## Is the dashboard fast when you view it in Metabase but slow when you've embedded it?
+
+**Root cause:** You can embed the charts and dashboards you create in Metabase in other websites in [several ways][embedding]. If people have to authenticate in order to view the dashboard in that other website, then you can pass their credentials to Metabase and use them to filter data. If they don't have to authenticate, though, then you can't do that, which means you may return a larger result set than you expect.
+
+**Steps to take:** This is almost always a result of a design error---if are only allowed to see a subset of the data when they're in Metabase, they shouldn't be able to see more (or all) of the data when the same question is displayed in an external web page. Please read [the introduction to embedding][embedding] to see how you can pass credentials from your site to Metabase.
+
+## Is Metabase and/or the data warehouse running on under-powered hardware?
+
+**Root cause:** If you run Metabase or the underlying data warehouse on a ten-year-old machine with only 1 GByte of RAM, it may not be able to keep up with your demands.
+
+**Steps to take:**
+
+1. Check the performance logs for the server where the database is running to see if it is hitting CPU or memory limits.
+2. Check the performance logs for the server where Metabase is running.
+3. If either is a bottleneck, upgrade to a more powerful server or one with more memory. If you would like this taken care of you, along with backups, cache configuration, and so on, please consider using [Metabase Cloud][metabase-start].
+
```

Comment:
one additional one to consider is for cases where a dashboard starts to slow down
1. did data volume grow? -> throw more hardware at it
2. are there more users on the data warehouse -> provision or set up quotas
3. are your execution slots in a database being saturated -> increase available execution slots

Reply:
agree with 1, but for 2 and 3 I would include something like: "ask your infrastructure team if there is any enqueuing being performed at the DH, if there is, ask them to raise the priority of Metabase queries", to make it more general for all the data sources we support

Reply:
New section added at the top.